### PR TITLE
Get project zone from metadata.google.internal

### PR DESCRIPTION
**Pull request by ArneGudermann**
Get now the project zone from the google metadata server.

https://cloud.google.com/appengine/docs/standard/python3/runtime#metadata_server

### CODE REVIEW

File: core/tasks.py [hunk 1]
```diff
@@ -64,14 +65,24 @@ def jsonDecodeObjectHook(obj):
...
-    queueRegion = regionMap.get(regionPrefix)
+
+    try:
+        headers = {"Metadata-Flavor": "Google"}
+        r = requests.get("http://metadata.google.internal/computeMetadata/v1/instance/zone", headers=headers)
+        # r.text should be look like this "projects/(project-number)/zones/(zone)-1"
+        # like so "projects/1234567890/zones/europe-west3-1"
+        queueRegion = "-".join(r.text.split("/")[-1].split("-")[:2])
+    except:  # Something went wrong with the Google Metadata Sever we use the old way
+        logging.warning("Can't get the queueRegion from the MetaData Server")
```
Comment (phorward) [inline core/tasks.py:79]: Can you please provide a hint why the connection to MetaData server failed?

Here's a suggestion:
```suggestion
    except Exception as e:  # Something went wrong with the Google Metadata Sever we use the old way
        logging.warning(f"Can't obtain queueRegion from Google MetaData Server due exception {e=}")
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann,
thank you, this is an interesting approach, and as long as we resist on Google App Engine I think this is useful. Anyway, we should discuss it in the VIUR meeting on friday. Nevertheless, please have a look at my suggestion.

**Review by phorward — APPROVED** [review]
I'm fine with it, but we should wait until friday. Thanks!

File: core/tasks.py [hunk 2]
```diff
@@ -64,14 +65,24 @@ def jsonDecodeObjectHook(obj):
...
 if _gaeApp:
-    regionPrefix = _gaeApp.split("~")[0]
-    queueRegion = regionMap.get(regionPrefix)
+
+    try:
+        headers = {"Metadata-Flavor": "Google"}
+        r = requests.get("http://metadata.google.internal/computeMetadata/v1/instance/zone", headers=headers)
+        # r.text should be look like this "projects/(project-number)/zones/(zone)-1"
+        # like so "projects/1234567890/zones/europe-west3-1"
+        queueRegion = "-".join(r.text.split("/")[-1].split("-")[:2])
```
Comment (sveneberth) [inline core/tasks.py:77]: You can save the join:
```suggestion
        queueRegion = r.text.split("/")[-1].rsplit("-", 1)[0]
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by phorward — APPROVED** [review]

**Pull request merged**